code/ship/shiphit.cpp

```diff
@@ -953,9 +957,9 @@ float do_subobj_hit_stuff(object *ship_objp, object *other_obj, vec3d *hitpos, i
 
 			// if this subsystem doesn't carry damage then subtract it off of our total return
 			if (subsystem->system_info->flags[Model::Subsystem_Flags::Carry_no_damage]) {
-				if ((other_obj == nullptr) || (other_obj->type != OBJ_SHOCKWAVE) || (!(subsystem->system_info->flags[Model::Subsystem_Flags::Carry_shockwave]))) {
+				if (other_obj_is_shockwave || (!(subsystem->system_info->flags[Model::Subsystem_Flags::Carry_shockwave]))) {
```

Comment:
This should be `!other_obj_is_shockwave`

Optional: remove the outer pair of parentheses from the second half of the condition to make it more readable